extensions-contrib/kubernetes-overlord-extensions/src/test/java/org/apache/druid/k8s/overlord/KubernetesTaskRunnerFactoryTest.java

```diff
@@ -0,0 +1,161 @@
+import org.apache.druid.server.DruidNode;
+import org.apache.druid.server.log.StartupLoggingConfig;
+import org.apache.druid.tasklogs.NoopTaskLogs;
+import org.apache.druid.tasklogs.TaskLogPusher;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class KubernetesTaskRunnerFactoryTest
+{
+  private ObjectMapper objectMapper;
+  private KubernetesTaskRunnerConfig kubernetesTaskRunnerConfig;
+  private StartupLoggingConfig startupLoggingConfig;
+  private TaskQueueConfig taskQueueConfig;
+  private TaskLogPusher taskLogPusher;
+  private DruidNode druidNode;
+  private TaskConfig taskConfig;
+
+  @Before
+  public void setup()
+  {
+    objectMapper = new TestUtils().getTestObjectMapper();
+    kubernetesTaskRunnerConfig = new KubernetesTaskRunnerConfig();
+    startupLoggingConfig = new StartupLoggingConfig();
+    taskQueueConfig = new TaskQueueConfig(
+        1,
+        null,
+        null,
+        null
+    );
+    taskLogPusher = new NoopTaskLogs();
+    druidNode = new DruidNode(
+        "test",
+        "",
+        false,
+        0,
+        1,
+        true,
+        false
+    );
+    taskConfig = new TaskConfig(
+        "/tmp",
+        null,
+        null,
+        null,
+        null,
+        false,
+        null,
+        null,
+        null,
+        false,
+        false,
+        null,
+        null,
+        false,
+        ImmutableList.of("/tmp")
+    );
+  }
+
+  @Test
+  public void test_get_returnsSameKuberentesTaskRunner_asBuild()
+  {
+    KubernetesTaskRunnerFactory factory = new KubernetesTaskRunnerFactory(
+        objectMapper,
+        kubernetesTaskRunnerConfig,
+        startupLoggingConfig,
+        taskQueueConfig,
+        taskLogPusher,
+        druidNode,
+        taskConfig
+    );
+
+    KubernetesTaskRunner expectedRunner = factory.build();
+    KubernetesTaskRunner actualRunner = factory.get();
+
+    Assert.assertEquals(expectedRunner, actualRunner);
+  }
+
+  @Test
+  public void test_build_withoutSidecarSupport_returnsKubernetesTaskRunnerWithSingleContainerTaskAdapter()
+  {
+    KubernetesTaskRunnerFactory factory = new KubernetesTaskRunnerFactory(
+        objectMapper,
+        kubernetesTaskRunnerConfig,
+        startupLoggingConfig,
+        taskQueueConfig,
+        taskLogPusher,
+        druidNode,
+        taskConfig
+    );
+
+    KubernetesTaskRunner runner = factory.build();
+    Assert.assertNotNull(runner);
+  }
+
+  @Test
+  public void test_build_withSidecarSupport_returnsKubernetesTaskRunnerWithMultiContainerTaskAdapter()
+  {
+    kubernetesTaskRunnerConfig.sidecarSupport = true;
+
+    KubernetesTaskRunnerFactory factory = new KubernetesTaskRunnerFactory(
+        objectMapper,
+        kubernetesTaskRunnerConfig,
+        startupLoggingConfig,
+        taskQueueConfig,
+        taskLogPusher,
+        druidNode,
+        taskConfig
+    );
+
+    KubernetesTaskRunner runner = factory.build();
+
+    Assert.assertNotNull(runner);
```

Comment:
you are verifying that the runner is not null. but it doesn't verify that the runner has a multi-container task adapter.